include/revng/ABIAnalyses/Common.h

```diff
@@ -0,0 +1,213 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/SmallSet.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/GlobalVariable.h"
+#include "llvm/IR/Instruction.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Value.h"
+#include "llvm/Support/Casting.h"
+
+#include "revng/BasicAnalyses/GeneratedCodeBasicInfo.h"
+#include "revng/MFP/MFP.h"
+#include "revng/Model/Binary.h"
+#include "revng/StackAnalysis/StackAnalysis.h"
+#include "revng/Support/revng.h"
+
+namespace ABIAnalyses {
+
+using llvm::BasicBlock;
+using llvm::CallInst;
+using llvm::GlobalVariable;
+using llvm::Instruction;
+using llvm::LoadInst;
+using llvm::SmallSet;
+using llvm::SmallVector;
+using llvm::StoreInst;
+using llvm::Value;
+
+using Register = model::Register::Values;
+
+enum TransferKind {
+  Read,
+  Write,
+  WeakWrite,
+  TheCall,
+  None,
+};
+
+struct ABIAnalysis {
+private:
+  SmallSet<GlobalVariable *, 20> ABIRegisters;
+  SmallVector<GlobalVariable *, 20> RegisterList;
+  const Instruction *CallSite;
+
+public:
+  ABIAnalysis(const GeneratedCodeBasicInfo &GCBI) :
+    ABIAnalysis(nullptr, GCBI){};
```

Comment:
Space before `{`. Are you using `clang-format`?

Reply:
it's actually `check-revng-conventions` that removes the space

Reply:
OK, `check-revng-conventions` is always right.